### DISCUSSION TITLE
portainer network problem

### DISCUSSION THREAD

**Discussion by gamebeaker** [firefly-iii/firefly-iii#9019] [Firefly III] [answered]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [ ] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I try to deploy firefly on my raspberrypi 4b.
The ip adresses from my firefly container are in the subnet 172.19.0.0/16 but my devices are on 192.168.0.0/24.
How can i change the subnet?

### Debug information

![grafik](https://github.com/firefly-iii/firefly-iii/assets/9269847/f6f06f13-990f-4405-8edf-94da99ff24e6)
![grafik](https://github.com/firefly-iii/firefly-iii/assets/9269847/0ea08fc6-fa3c-4c9c-8532-02a8e3631205)

**Comment by gamebeaker** [comment]
My solution was that i had a macvlan network in docker and i changed all container networks to my macvlan and deleted the original firefly network from the containers.
![grafik](https://github.com/firefly-iii/firefly-iii/assets/9269847/5dc492dc-0f99-4908-999e-2506a50e645e)
What is a MACVLAN? https://youtu.be/bKFMS5C4CG0?si=fc85zICUsIGragde&t=940

**Comment by gamebeaker** [comment, marked as answer]
Better solution: (the ips jumped all over the place)
in the docker compose change
<pre>
    networks:
      firefly_iii:
</pre>
to
<pre>
    networks:
      firefly_iii:
        ipv4_address: 192.168.0.XXX
</pre>
for all services.
At the end 
<pre>
networks:
  firefly_iii:
    driver: bridge
</pre>
to
<pre>
networks:
  firefly_iii:
    name: MACVLAN (existing Network type macvlan) 
    external: true
</pre>

**Reply by JC5** [reply]
Nice!
